Flaky querier test with bucket client mock

Failed at: https://github.com/grafana/mimir/actions/runs/5291844667/jobs/9577945233?pr=5266
and at https://github.com/grafana/mimir/actions/runs/5291869360/jobs/9578004700?pr=3555

```
level=debug user=1 component=block.MetaFetcher msg="fetching meta data" concurrency=1
level=info user=1 component=block.MetaFetcher msg="successfully synchronized block metadata" duration=1.00077898s duration_ms=1000 cached=0 returned=0 partial=0
level=error msg="unable to run the initial blocks scan" err="context canceled"
level=debug user=user-1 component=block.MetaFetcher msg="fetching meta data" concurrency=1
panic: 
assert: mock: I don't know what to return because the method call was unexpected.
	Either do Mock.On("Get").Return(...) first, or remove the Get() call.
	This method was unexpected:
		Get(*context.cancelCtx,string)
		0: &context.cancelCtx{Context:(*context.emptyCtx)(0xc00004e098), mu:sync.Mutex{state:0, sema:0x0}, done:atomic.Value{v:(chan struct {})(0xc00066f8c0)}, children:map[context.canceler]struct {}(nil), err:(*errors.errorString)(0xc0000[54](https://github.com/grafana/mimir/actions/runs/5291844667/jobs/9577945233?pr=5266#step:8:55)300), cause:(*errors.errorString)(0xc000054300)}
		1: "user-1/00000000010000000000000000/meta.json"
	at: [/__w/mimir/mimir/pkg/storage/bucket/client_mock.go:79 /__w/mimir/mimir/pkg/storage/bucket/prefixed_bucket_client.go:63 /__w/mimir/mimir/pkg/storage/bucket/sse_bucket_client.go:109 /__w/mimir/mimir/pkg/storage/tsdb/block/fetcher.go:227 /__w/mimir/mimir/pkg/storage/tsdb/block/fetcher.go:305 /__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:75 /usr/local/go/src/runtime/asm_amd64.s:1598]

goroutine 331 [running]:
github.com/stretchr/testify/mock.(*Mock).fail(0xc0006b03c0, {0x22a6d21, 0xc1}, {0xc000704900, 0x4, 0x4})
	/__w/mimir/mimir/vendor/github.com/stretchr/testify/mock/mock.go:332 +0x1db
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0006b03c0, {0x36fe451, 0x3}, {0xc00081fa60, 0x2, 0x2})
	/__w/mimir/mimir/vendor/github.com/stretchr/testify/mock/mock.go:499 +0x8eb
github.com/stretchr/testify/mock.(*Mock).Called(0xc00066d680?, {0xc00081fa60, 0x2, 0x2})
	/__w/mimir/mimir/vendor/github.com/stretchr/testify/mock/mock.go:464 +0x1a5
github.com/grafana/mimir/pkg/storage/bucket.(*ClientMock).Get(0xc0006b03c0, {0x33e6a50?, 0xc0006b0410}, {0xc00066d680, 0x2b})
	/__w/mimir/mimir/pkg/storage/bucket/client_mock.go:79 +0x16a
github.com/grafana/mimir/pkg/storage/bucket.(*PrefixedBucketClient).Get(0xc00081f820, {0x33e6a50, 0xc0006b0410}, {0xc00066d650, 0x24})
	/__w/mimir/mimir/pkg/storage/bucket/prefixed_bucket_client.go:63 +0xd4
github.com/grafana/mimir/pkg/storage/bucket.(*SSEBucketClient).Get(0xc00078[56](https://github.com/grafana/mimir/actions/runs/5291844667/jobs/9577945233?pr=5266#step:8:57)e0, {0x33e6a50, 0xc0006b0410}, {0xc00066d[65](https://github.com/grafana/mimir/actions/runs/5291844667/jobs/9577945233?pr=5266#step:8:66)0, 0x24})
	/__w/mimir/mimir/pkg/storage/bucket/sse_bucket_client.go:109 +0x78
github.com/grafana/mimir/pkg/storage/tsdb/block.(*MetaFetcher).loadMeta(0xc0006c8780, {0x33e6a50, 0xc0006b0410}, {0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, ...})
	/__w/mimir/mimir/pkg/storage/tsdb/block/fetcher.go:227 +0xb0f
github.com/grafana/mimir/pkg/storage/tsdb/block.(*MetaFetcher).fetchMetadata.func1()
	/__w/mimir/mimir/pkg/storage/tsdb/block/fetcher.go:305 +0xc8
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:75 +0x83
created by golang.org/x/sync/errgroup.(*Group).Go
	/__w/mimir/mimir/vendor/golang.org/x/sync/errgroup/errgroup.go:[72](https://github.com/grafana/mimir/actions/runs/5291844667/jobs/9577945233?pr=5266#step:8:73) +0x12f
FAIL	github.com/grafana/mimir/pkg/querier	23.306s
```

--- a/pkg/querier/blocks_finder_bucket_scan_test.go
+++ b/pkg/querier/blocks_finder_bucket_scan_test.go
@@ -168,20 +168,23 @@ func TestBucketScanBlocksFinder_StopWhileRunningTheInitialScanOnManyBlocks(t *te
 	}
 
 	// Mock the bucket to introduce a 1s sleep while syncing each block in the bucket.
-	bucket := &bucket.ClientMock{}
-	bucket.MockIter("", []string{"user-1"}, nil)
-	bucket.MockIter("user-1/", blockPaths, nil)
-	bucket.On("Exists", mock.Anything, mock.Anything).Return(false, nil).Run(func(args mock.Arguments) {
-		// We return the meta.json doesn't exist, but introduce a 1s delay for each call.
+	bkt := &bucket.ClientMock{}
+	bkt.MockIter("", []string{"user-1"}, nil)
+	bkt.MockIter("user-1/", blockPaths, nil)
+
+	// We return that all files don't exist, but introduce a 1s delay for each call.
+	sleep := func(_ mock.Arguments) {
 		time.Sleep(time.Second)
-	})
+	}
+	bkt.On("Exists", mock.Anything, mock.Anything).Return(false, nil).Run(sleep)
+	bkt.On("Get", mock.Anything, mock.Anything).Return(nil, bucket.ErrObjectDoesNotExist).Run(sleep)
 
 	cfg := prepareBucketScanBlocksFinderConfig()
 	cfg.CacheDir = t.TempDir()
 	cfg.MetasConcurrency = 1
 	cfg.TenantsConcurrency = 1
 
-	s := NewBucketScanBlocksFinder(cfg, bucket, nil, log.NewLogfmtLogger(os.Stdout), nil)
+	s := NewBucketScanBlocksFinder(cfg, bkt, nil, log.NewLogfmtLogger(os.Stdout), nil)
 
 	// Start the scanner, let it run for 1s and then issue a stop.
 	require.NoError(t, s.StartAsync(context.Background()))

